src/MSBuildLocator/MSBuildLocator.Mono.cs

```diff
@@ -0,0 +1,144 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+
+namespace Microsoft.Build.Locator
+{
+    public static partial class MSBuildLocator
+    {
+        private static readonly string s_monoMSBuildDll_Current_RelativePath = Path.Combine ("lib", "mono", "msbuild", "Current", "bin", "MSBuild.dll");
+        private static readonly string s_monoMSBuildDll_15_0_RelativePath    = Path.Combine ("lib", "mono", "msbuild", "15.0", "bin", "MSBuild.dll");
+        private static readonly string s_monoOSXBasePath = "/Library/Frameworks/Mono.framework/Versions";
+
+        internal static IEnumerable<VisualStudioInstance> GetMonoMSBuildInstances()
+        {
+            // $prefix/lib/mono/4.5/mscorlib.dll
+            var runningMonoFullPath = Path.GetFullPath(Path.Combine(Path.GetDirectoryName(typeof (object).Assembly.Location), "..", "..", ".."));
+            if (TryGetValidMonoVersion(runningMonoFullPath, out var version))
+            {
+                yield return new VisualStudioInstance("Mono", runningMonoFullPath, version, DiscoveryType.Mono);
+            }
+
+            if (!IsOSX)
+            {
+                // Returning just one instance on !osx
+                yield break;
+            }
+
+            foreach(var dirPath in Directory.EnumerateDirectories(s_monoOSXBasePath))
+            {
+                if (string.Equals(Path.GetFileName(dirPath), "Current") || // skip the `Current` symlink
+                    string.Equals(Path.GetFullPath(dirPath), runningMonoFullPath)) // and the running mono version
+                {
+                    continue;
+                }
+
+                if (TryGetValidMonoVersion(dirPath, out version))
+                {
+                    yield return new VisualStudioInstance("Mono", dirPath, version, DiscoveryType.Mono);
+                }
+            }
+
+            bool TryGetValidMonoVersion(string path, out Version ver)
+            {
+                ver = null;
+                if (!File.Exists(Path.Combine(path, s_monoMSBuildDll_Current_RelativePath)) &&
+                        !File.Exists(Path.Combine(path, s_monoMSBuildDll_15_0_RelativePath)))
+                {
+                    return false;
+                }
+
+                if (TryGetMonoVersionFromMonoBinary(path, out ver) || Version.TryParse(Path.GetFileName(path), out ver))
+                {
+                    return true;
+                }
+
+                // The path has a valid mono, but we can't find the version
+                // so, let's return the instance at least but with version 0.0.0
+                ver = new Version(0, 0, 0);
```

Comment:
Not sure whether this is better or it's better to not return it. Users could find it separately and register it anyway. @rainersigwald, thoughts?

Reply:
Ok to merge? Or are we waiting for this?

Reply:
I'd prefer to wait for @rainersigwald's signoff. Also, would you mind adding the if-not-mono check discussed [here](https://github.com/microsoft/MSBuildLocator/pull/82#issuecomment-754129837)?

Reply:
Added, and fixed a bug introduced in the nuget PR.

Reply:
This is probably ok, let's just listen for feedback.